shiny/ui/_include_helpers.py

```diff
@@ -265,4 +338,111 @@ def read_utf8(path: Path | str) -> str:
         return f.read()
 
 
+def check_bootstrap_compatibility(
+    theme: Path | str | HTMLDependency,
+    bs_version: Optional[Version | str] = None,
+) -> None:
+    if isinstance(theme, HTMLDependency):
+        check_bootstrap_compatibility_dep(theme, bs_version)
+        return
+
+    if bs_version is not None:
+        check_bootstrap_compatibility_version(bs_version)
+        return
+
+    if isinstance(theme, (Path, str)):
+        check_bootstrap_compatibility_file(theme)
+        return
+
+    raise RuntimeError(
+        "Invalid theme argument. Must be a Path, str, or HTMLDependency."
+    )
+
+
+def check_bootstrap_compatibility_dep(
+    theme: HTMLDependency,
+    bs_version: Optional[Version | str] = None,
+) -> None:
+    if bs_version is not None:
+        check_bootstrap_compatibility_version(
+            bs_version=bs_version,
+            name=theme.name,
+            version=theme.version,
+        )
+
+    if hasattr(theme, "bs_version"):
```

Comment:
I think some of my concerns about complexity here could go away if we decided we didn't need to introduce yet another representation of the version